### PR TITLE
Add email formatter

### PR DESCRIPTION
**Pull request by vadymmarkov**
Related to https://github.com/hyperoslo/Form/issues/357

### CODE REVIEW

File: Source/Formatters/FORMEmailFormatter.m [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#import "FORMEmailFormatter.h"
+
+@implementation FORMEmailFormatter
+
+- (NSString *)formatString:(NSString *)string reverse:(BOOL)reverse {
+    string = [super formatString:string reverse:reverse];
+    if (!string) return nil;
```
Comment (zenangst) [inline Source/Formatters/FORMEmailFormatter.m]: I guess we should start using braces for all `if` statements now.

Comment (vadymmarkov) [inline Source/Formatters/FORMEmailFormatter.m]: @zenangst Fixed.

File: Tests/Tests/Formatters/FORMEmailFormatterTests.m [hunk 2]
```diff
@@ -0,0 +1,24 @@
+//
```
Comment (zenangst) [inline Tests/Tests/Formatters/FORMEmailFormatterTests.m]: We tend to remove all header comments blocks ;)

Comment (vadymmarkov) [inline Tests/Tests/Formatters/FORMEmailFormatterTests.m]: @zenangst Right, my bad.

File: Tests/Tests/Formatters/FORMEmailFormatterTests.m [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+@interface FORMEmailFormatterTests : XCTestCase
+
+@end
+
+@implementation FORMEmailFormatterTests
+
+- (void)testFormatString {
+    FORMEmailFormatter *formatter = [FORMEmailFormatter new];
+    NSString *inputString = @"t,e,st@example.com";
+    XCTAssertTrue([@"test@example.com" isEqualToString:[formatter formatString:inputString reverse:NO]]);
```
Comment (3lvis) [inline Tests/Tests/Formatters/FORMEmailFormatterTests.m]: Might be better if you do:

```objc
XCTAssertEqualObjects(@"test@example.com", [formatter formatString:inputString reverse:NO]);
```

Comment (3lvis) [inline Tests/Tests/Formatters/FORMEmailFormatterTests.m]: Or even 

```objc
XCTAssertEqualObjects(@"test@example.com", [formatter formatString:@"t,e,st@example.com" reverse:NO]);
```

**Comment by 3lvis** [comment]
Great job @vadymmarkov!

**Pull request merged**